course/api/views.py

```diff
@@ -68,6 +68,20 @@ def get_object(self):
         return self.get_member_object('module', 'Exercise module')
 
 
+class CourseExerciseTreeViewSet(CourseResourceMixin,
+                                viewsets.ViewSet):
+    serializer_class = TreeCourseModuleSerializer
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.serializer_class(
+            self.content.data['modules'],
```

Comment:
Just writing down that `self.content` comes from the super class `CourseResouceMixin - CourseInstanceBaseMixin`. It is the course cache `CachedContent`.

* https://github.com/apluslms/a-plus/blob/46386ac304ebfb8dc00bfd2bffdb7d0d01d6ae8a/course/viewbase.py#L61
* https://github.com/apluslms/a-plus/blob/46386ac304ebfb8dc00bfd2bffdb7d0d01d6ae8a/exercise/cache/content.py#L10